libraries/tools/kotlin-symbol-processing-api/src/org/jetbrains/kotlin/ksp/symbol/Origin.kt

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.ksp.symbol
+
+enum class Origin {
+    KOTLIN,
+    CLASS,
```

Comment:
With the presence of Kotlin metadata, do we need `CLASS`?

Reply:
We do still want to distinguish between source and library, the reason for that is because if a symbol is from source we need additional effort to compute implicit logics like default visibility, whereas a symbol from binary have everything ready.

Reply:
Actually there is PACKAGE_VISIBILITY in descriptor visibilities, my mistake that I didn't include them in my change.